core/assets/shaders/shield.frag

```diff
@@ -28,8 +27,23 @@ void main(){
 	if(texture2D(u_texture, T).a < 0.9 &&
        		(texture2D(u_texture, T + vec2(0, step) * v).a > 0.0 || texture2D(u_texture, T + vec2(0, -step) * v).a > 0.0 ||
        		texture2D(u_texture, T + vec2(step, 0) * v).a > 0.0 || texture2D(u_texture, T + vec2(-step, 0) * v).a > 0.0)){
-
-		gl_FragColor = mix(u_shieldcolor, vec4(1.0), si);
+                
+		vec4 bordercolor = vec4(0.0, 0.0, 0.0, 0.0);
+
+		if(texture2D(u_texture, T + vec2(0, step) * v).a > 0.9){
```

Comment:
Why are the statements split up this way?

Reply:
because I focused on your code and code from other shaders and highlighted the variable Declaration for readability.

Reply:
Don't quite understand the claim to margins, what's wrong with them?